### PR TITLE
Scoped context

### PR DESCRIPTION
**Pull request by dlyr**
* **Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
Add scoped context behavior. One can ask WindowQt an OpengGLContext than is bound at variable creation, and unbound when the variable is deleted (end of scope).

* **What is the current behavior?** (You can also link to an open issue here)
Have to manage context activation with makeCurrent and doneCurrent.
In a block, some gl related variable might be deleted on block end, while the doneCurrent have to be called before block end explicitly, see #912 .

* **What is the new behavior (if this is a feature change)?**
Such block binded context are obtained as an ScopedContext object with WindowQt::createScopedContext, ScopedContext ctor calls makeCurrent, while dtor calls doneCurrent. 
Hence the "make current count" is still in use, and multiple ScopedContext can be imbricated without issues.

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)
Nope, but this new behavior can replace makeCurrent/doneCurrent in a futur PR.

* **Other information**:
Also fix possible segfault on cube texture gpu allocation when no data provided

### CODE REVIEW

File: src/Engine/Data/Texture.cpp [hunk 1]
```diff
@@ -132,7 +132,14 @@ void Texture::updateData( const void* data ) {
     } break;
     case GL_TEXTURE_CUBE_MAP: {
         // Load the 6 faces of the cube-map
-        void** texels = (void**)data;
+        void* nullTexels[6] { nullptr, nullptr, nullptr, nullptr, nullptr, nullptr };
```
Comment (nmellado) [inline src/Engine/Data/Texture.cpp:135]: Maybe use 
```suggestion
        static const void* nullTexels[6] { nullptr, nullptr, nullptr, nullptr, nullptr, nullptr };
```
instead ?

File: src/Gui/Viewer/WindowQt.inl [hunk 2]
```diff
@@ -0,0 +1,22 @@
+#pragma once
+#include <Gui/Viewer/WindowQt.hpp>
+
+namespace Ra {
+namespace Gui {
+
+class WindowQt::ScopedContext
```
Comment (nmellado) [inline src/Gui/Viewer/WindowQt.inl:7]: Question about the naming: is this necessary related to GL ? If yes, it could be named `ScopedGLContext`.

File: src/Engine/Data/Texture.cpp [hunk 3]
```diff
@@ -132,7 +132,14 @@ void Texture::updateData( const void* data ) {
...
+        void** texels;
+
+        if ( data == nullptr )
+            texels = (void**)data;
+        else
+            texels = nullTexels;
```
Comment (hiergaut) [inline src/Engine/Data/Texture.cpp:141]:
```suggestion
        if ( data == nullptr )
            texels = nullTexels;
        else
            texels = (void**)data;
```

File: src/Gui/Viewer/WindowQt.inl (hunk 2 above)
Comment (dlyr) [reply src/Gui/Viewer/WindowQt.inl:7]: Yes it's related to GL. Do you suggest to rename the inner class only or also the activateScopedContext method ?
Other Qt/GL related fonction are named context, makeCurrent and doneCurrent, following the Qt's convention. 
So my first opinion is ok to rename inner class, keep method name

File: src/Engine/Data/Texture.cpp (hunk 3 above)
Comment (dlyr) [reply src/Engine/Data/Texture.cpp]: moved to #904

File: src/Engine/Data/Texture.cpp (hunk 1 above)
Comment (dlyr) [reply src/Engine/Data/Texture.cpp]: moved to #904

File: src/Gui/Viewer/WindowQt.inl (hunk 2 above)
Comment (nmellado) [reply src/Gui/Viewer/WindowQt.inl]: Ok, sounds good, I guess this will be more readable.

**Review by nmellado — APPROVED** [review]
Ready to merge once updated wrt `release_candidate`
